File: src/main/java/net/softbell/bsh/dto/view/advance/TriggerSummaryCardDto.java
package net.softbell.bsh.dto.view.advance;

import lombok.Getter;
import lombok.Setter;
import net.softbell.bsh.domain.EnableStatusRule;
import net.softbell.bsh.domain.entity.NodeTrigger;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 트리거뷰 카드정보 DTO
 */
@Getter
@Setter
public class TriggerSummaryCardDto
{
	private Long triggerId;
	private String description;
	private boolean enableStatus;
	private String creatorNickname;
	
	public TriggerSummaryCardDto(NodeTrigger entity)
	{
		// Exception
		if (entity == null)
			return;
		
		// Convert
		this.triggerId = entity.getTriggerId();
		this.description = entity.getDescription();
		
		if (entity.getEnableStatus() == EnableStatusRule.ENABLE)
			this.enableStatus = true;
		else
			this.enableStatus = false;
		this.creatorNickname = entity.getMember().getNickname();
	}
}

File: src/main/kotlin/net/softbell/bsh/dto/view/advance/TriggerSummaryCardDto.kt
package net.softbell.bsh.dto.view.advance

import net.softbell.bsh.domain.EnableStatusRule
import net.softbell.bsh.domain.entity.NodeTrigger

/**
 * @author : Bell(bell@softbell.net)
 * @description : 트리거뷰 카드정보 DTO
 */
class TriggerSummaryCardDto(entity: NodeTrigger) {
    val triggerId: Long = entity.triggerId
    val description: String = entity.description
    val enableStatus = entity.enableStatus == EnableStatusRule.ENABLE
    val creatorNickname: String = entity.member.nickname
}